Deduplicate sections WSTG-CONF-06 and WSTG-INPV-03

johanna-a opened the pull request:
Sections WSTG-CONF-06 (Test HTTP Methods) and WSTG-INPV-03 (Testing for HTTP
Verb Tampering) are consolidated under WSTG-CONF-06.

This PR covers issue #463 .

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- The sections WSTG-CONF-06 (Test HTTP Methods) and WSTG-INPV-03 (Testing for HTTP
Verb Tampering) were very similar in content. After a discussion (see issue #463) it was decided to consolidate these under WSTG-CONF-06

johanna-a commented:
As this is my first PR against OWASP/wstg, please have patience with me :)

I didn't know how to handle the renumbering of sections in WSTG-INPV, so I left that for now instead of making a misguided attempt.

kingthorin commented:
Thanks skimmed it quickly seems good to me. Will do a full review in a few hours.

kingthorin commented on document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:
```suggestion
Oddly, the other valid HTTP methods (mentioned earlier) are not supported by the [HTML standard](https://www.w3.org/TR/REC-html40/interact/forms.html#h-17.13.1). Any HTTP method other than GET or POST needs to be called outside the HTML document. However, JavaScript and AJAX calls may send methods other than GET and POST. For this reason, most web applications should not need to accept other HTTP methods.
```

kingthorin commented:
I think we'll just renumber everything, which is a bit of a pain in the rear. But, it's also addressed by other recent content added to the main readme and a leadin section which covers how the WSTG should be referenced.

johanna-a commented:
> I think we'll just renumber everything, which is a bit of a pain in the rear. But, it's also addressed by other recent content added to the main readme and a leadin section which covers how the WSTG should be referenced.
Renumbering should just be a few lines of emacs-lisping or python-ing away, right? ;)
Do you want to look into it? Anyhow I think it's a good idea to put it in a different PR for clarity.
> If there's debate or a want for a shorter timelines, you could just put a placeholder in the file instead of removing it (as a temporary measure).
This is a good idea, I updated the PR.

ThunderSon commented on document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:
`HTML standard` is a wrong name for this link. This is discussing Forms, specifically. Kindly update it to reflect what you're pointing at.

johanna-a replied:
I removed this sentence altogether, I think it actually didn't add much valuable information.

ThunderSon commented on document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:
Can we re-word this to something more on the lines of "Based on the needs, setup the methods". The way it was discussed, and is still discussed, almost all services for example NEED OPTIONS, and HEAD is a smaller GET, which is used by bots. Methods are simply like any other part of the web page. If it doesn't belong to it, remove it. HTTP methods are not some special evil that needs shutting down. Can we try and stop conveying that?

ThunderSon replied:
And in the add try to set a thing for actions. CSRF based attacks, unauthenticated, hidden, etc.

ThunderSon commented on document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:
```diff
@@ -6,26 +6,36 @@
-HTTP offers a number of methods that can be used to perform actions on the web server. Many of theses methods are designed to aid developers in deploying and testing HTTP applications. These HTTP methods can be used for nefarious purposes if the web server is misconfigured. Additionally, Cross Site Tracing (XST), a form of cross site scripting using the server's HTTP TRACE method, is examined.
-While GET and POST are by far the most common methods that are used to access information provided by a web server, the Hypertext Transfer Protocol (HTTP) allows several other (and somewhat less known) methods. RFC 2616 (which describes HTTP version 1.1 which is the standard today) defines the following eight methods:
+HTTP offers a number of methods that can be used to perform actions on the web server (the HTTP 1.1 standard refers to them as `methods` but they are also commonly described as `verbs`). Many of theses methods are designed to aid developers in deploying and testing HTTP applications. These HTTP methods can be used for nefarious purposes if the web server is misconfigured. Additionally, Cross Site ...
+While GET and POST are by far the most common methods that are used to access information provided by a web server, the Hypertext Transfer Protocol (HTTP) allows several other (and somewhat less known) methods.
```
HTTP was already used at the start, so either use it in here directly, or expand it at the start.

johanna-a replied:
Agree, will change in next update of the PR.

ThunderSon commented on document/4-Web_Application_Security_Testing/02-Configuration_and_Deployment_Management_Testing/06-Test_HTTP_Methods.md:
```diff
@@ -6,26 +6,36 @@
 
-HTTP offers a number of methods that can be used to perform actions on the web server. Many of theses methods are designed to aid developers in deploying and testing HTTP applications. These HTTP methods can be used for nefarious purposes if the web server is misconfigured. Additionally, Cross Site Tracing (XST), a form of cross site scripting using the server's HTTP TRACE method, is examined.
-While GET and POST are by far the most common methods that are used to access information provided by a web server, the Hypertext Transfer Protocol (HTTP) allows several other (and somewhat less known) methods. RFC 2616 (which describes HTTP version 1.1 which is the standard today) defines the following eight methods:
+HTTP offers a number of methods that can be used to perform actions on the web server (the HTTP 1.1 standard refers to them as `methods` but they are also commonly described as `verbs`). Many of theses methods are designed to aid developers in deploying and testing HTTP applications. These HTTP methods can be used for nefarious purposes if the web server is misconfigured. Additionally, Cross Site ...
```
Link to this as well for `Cross Site Tracing` https://owasp.org/www-community/attacks/Cross_Site_Tracing

ThunderSon replied:
I wonder, should the XST attack be discussed on its own like this in the summary? @kingthorin 
It feels like being part of the discussion later on, instead of having a special place in the Summary.
Maybe discuss what are the possible attack vectors in the Summary.

kingthorin replied:
Ya I guess it doesn't really need to be called out here.

johanna-a replied:
I agree that the XST attack shouldn't be described on its own in the summary, so I've reworked that. I like the idea of a more general description of possible attack vectors, I'll try to write a draft of that and lets see how it works.

ThunderSon commented:
For the numbering, should we take the numbering directly, or wait before a version release? I believe we can keep the md file  in the latest but change the numbering to XXX inside the file. The naming convention of the files ... it could be improved based on this discussion, don't you think :wink:

ThunderSon commented:
@johanna-a is this up for review again?

johanna-a commented:
I'm realizing that both sections were in need of some rework and that combining them made this very apparent, so I pulled back this into the 'draft' stage and will continue to update this PR in the following week or two.

kingthorin commented:
@johanna-a how is this coming along?

johanna-a commented:
> @johanna-a how is this coming along?
I've had a bit much on my plate lately, so it's been on the backburner for a while... ^^;
Thanks for pinging me! I'll try to get some time to work on this shortly :)

kingthorin commented:
No problem, life happens, we get that. Let us know if you get stuck or there's something we can do to help.

ThunderSon commented:
This will be closed and work will proceed on #502 . @victoriadrake did you mean to commit instead of opening a new PR?
